src/metadata.rs

```diff
@@ -37,20 +49,31 @@ pub async fn run(schema: &Metadata) -> Result<()> {
     Ok(())
 }
 
-async fn fetch_package_config_from_file(app_instance: &str) -> Result<PackageConfig> {
+async fn fetch_package_config_from_file(
+    app_instance: &str,
+    skip_auth: bool,
+) -> Result<PackageConfig> {
     let file = File::open(app_instance)?;
     let app_instance: AppInstance = serde_yaml::from_reader(file)?;
-    fetch_package_config_local_auth(&app_instance).await
+    fetch_package_config_local_auth(&app_instance, skip_auth).await
 }
 
-pub async fn fetch_package_config_local_auth(app_instance: &AppInstance) -> Result<PackageConfig> {
+pub async fn fetch_package_config_local_auth(
+    app_instance: &AppInstance,
+    skip_auth: bool,
+) -> Result<PackageConfig> {
     let reference: Reference = app_instance.spec.package.image.parse()?;
-    let credentials = docker_credential::get_credential(reference.registry())?;
-    let DockerCredential::UsernamePassword(username, password) = credentials else {
-        bail!("unsupported docker credentials")
-    };
-    let auth = RegistryAuth::Basic(username, password);
 
-    let config = oci::fetch_package_config(app_instance, &auth).await?;
-    Ok(config)
+    if skip_auth {
+        let config = oci::fetch_package_config(app_instance, &RegistryAuth::Anonymous).await?;
+        Ok(config)
+    } else {
+        let credentials = docker_credential::get_credential(reference.registry())?;
+        let DockerCredential::UsernamePassword(username, password) = credentials else {
+            bail!("unsupported docker credentials")
+        };
+        let auth = RegistryAuth::Basic(username, password);
+        let config = oci::fetch_package_config(app_instance, &auth).await?;
+        Ok(config)
+    }
 }
```

Comment:
An entirely optional refactoring could be something like:
```
let auth = if skip_auth {
        RegistryAuth::Anonymous
    } else {
        let credentials = docker_credential::get_credential(reference.registry())?;
        let DockerCredential::UsernamePassword(username, password) = credentials else {
            bail!("unsupported docker credentials")
        };
        RegistryAuth::Basic(username, password)
    };
let config = oci::fetch_package_config(app_instance, &auth).await?;
Ok(config)
```

Reply:
Ah! I like this much more.

I had something similar to `let auth = if skip_auth ...` before, but it was below the original stuff and the logic was all wrong, intuitively I just moved it to the standard `if else` :smile: 

I'll change it to this because I think its much clearer :100: